### PR TITLE
Enable SPEC SAT for Java sources

### LINKED ISSUE #17580: [Spike] Adding SAT testing to Java connectors

### LINKED ISSUE #18556: Add SAT to all Java connectors and test just the SPEC command

### PR DESCRIPTION
**Pull request by DoNotPanicUA**
## What
Add missing files and configurations to run SAT SPEC as part of the integration tests.
:information_source: [Tech spec](https://docs.google.com/document/d/1551FZmMH5y8Nn1m6wy8_O3-yVNpdADBPAcAuZK1sHSU/edit#heading=h.5vsepilo0o5q)

:warning: Message for reviewers : 
1) The snowflake source passes all SATs but fails on invalid oath creds. It's not related to this PR, but I'm also working on fixing it.
2) The scaffold-java-jdbc is not implemented at all. There is no way to add these tests thoroughly.

| Database source | Enabled SAT SPEC | Succesfully pass tests | Fix spec format |
|---|---|---|---|
| bigquery | :heavy_check_mark: | :heavy_check_mark: | :heavy_minus_sign: |
| clickhouse | :heavy_check_mark: | :heavy_check_mark: | :heavy_minus_sign: |
| cockroachdb | :heavy_check_mark: | :heavy_check_mark: | :heavy_minus_sign: |
| db2 | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: |
| mssql | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: |
| mysql | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: |
| oracle | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: |
| postgres | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: |
| redshift | :heavy_check_mark: | :heavy_check_mark: | :heavy_minus_sign: |
| scaffold-java-jdbc | :x: | :x: | :x: |
| snowflake | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: |
| tidb | :heavy_check_mark: | :heavy_check_mark: | :heavy_minus_sign: |
| elasticsearch | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: |
| kafka | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: |
| mongodb-v2 | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: |
| sftp | :heavy_check_mark: | :heavy_check_mark: | :heavy_minus_sign: |

:warning: _`scaffold-java-jdbc` source is not implemented!_

### CODE REVIEW

**Comment by grubberr** [comment]
@DoNotPanicUA @sherifnada 

### Source acceptance test AND Integration tests mess in airbyte

I have counted 3 ways to run SAT tests:

1. [acceptance-test-docker.sh](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-github/acceptance-test-docker.sh)

2. `python -m pytest integration_tests -p integration_tests.acceptance` - from docs
    
3. `./gradlew airbyte-integrations:connectors:source-github::sourceAcceptanceTest` - [here](https://github.com/airbytehq/airbyte/blob/master/buildSrc/src/main/groovy/airbyte-source-acceptance-test.gradle#L22) - CI way

Ways (2),(3) - are running SAT through Integration tests, but way (1) - runs SAT directly.
That's way (1) can work for you but (2), (3) can fail.

**Comment by grubberr** [comment]
@DoNotPanicUA @sherifnada

to be more concise

For ways (2), (3) we need this file `integration_tests/acceptance.py`
but way (1) don't use this file `integration_tests/acceptance.py`

File: airbyte-integrations/connectors/source-bigquery/integration_tests/acceptance.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-bigquery/integration_tests/acceptance.py:1]: I don't think these `/integration_tests` files are needed

Comment (grubberr) [reply airbyte-integrations/connectors/source-bigquery/integration_tests/acceptance.py:1]: https://github.com/airbytehq/airbyte/blob/master/buildSrc/src/main/groovy/airbyte-source-acceptance-test.gradle#L22

?

Comment (DoNotPanicUA) [reply airbyte-integrations/connectors/source-bigquery/integration_tests/acceptance.py:1]: @evantahler 
if want to run the tests only locally as we see [here](https://github.com/airbytehq/airbyte/pull/16445), yes.
If you want to run it as part of the CI testing, you have to add it.

**Comment by DoNotPanicUA** [comment]
Add a new secret file for the SPEC tests with the `config.json` file.
Secret name `SECRET_SOURCE-BIGQUERY_CONFIG__CREDS`

File: airbyte-integrations/connectors/source-bigquery/acceptance-test-config.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# See [Source Acceptance Tests](https://docs.airbyte.com/connector-development/testing-connectors/source-acceptance-tests-reference)
+# for more information about how to configure these tests
+connector_image: airbyte/source-bigquery:dev
+tests:
+  spec:
+    - spec_path: "src/test-integration/resources/expected_spec.json"
+      config_path: "src/test-integration/resources/dummy_config.json"
```
Comment (grubberr) [inline airbyte-integrations/connectors/source-bigquery/acceptance-test-config.yml:7]: BTW in source python connectors we usually store dummy, invalid configs here:
`integration_tests/invalid_config.json`
it just FYI

Comment (DoNotPanicUA) [reply airbyte-integrations/connectors/source-bigquery/acceptance-test-config.yml:7]: thanks, this is a valid config, but mocked :)
we don't have any configs actually before we up test environment.

File: airbyte-integrations/connectors/source-bigquery/acceptance-test-docker.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# Pull latest acctest image
+docker pull airbyte/source-acceptance-test:latest
+
+# Run
+docker run --rm -it \
+    -v /var/run/docker.sock:/var/run/docker.sock \
+    -v /tmp:/tmp \
+    -v $(pwd):/test_input \
+    airbyte/source-acceptance-test \
+    --acceptance-test-config /test_input
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-bigquery/acceptance-test-docker.sh:15]: Please run `./gradlew format` to format your files

File: airbyte-integrations/connectors/source-bigquery/integration_tests/acceptance.py (hunk 1 above)
Comment (alafanechere) [reply airbyte-integrations/connectors/source-bigquery/integration_tests/acceptance.py:1]: The `acceptance.py` is indeed required to launch the test suite with the SAT plugin. We should find a better way because it's far from ideal. @DoNotPanicUA could you please create an issue about this?

**Review by alafanechere — CHANGES_REQUESTED** [review]
Thank you for this work!
I think two missing steps are required:
1. Format your code `./gradlew format`
2. If you update source-scaffold-java-jdbc in this PR you'll need to change the Java jdbc code generator to create the files you've added (https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/plopfile.js#L1). Feel free to do this in a separate PR but please remove changes on source-scaffold-java-jdbc to make the connectors base build pass.

File: airbyte-integrations/connectors/source-bigquery/acceptance-test-docker.sh (hunk 3 above)
Comment (DoNotPanicUA) [reply airbyte-integrations/connectors/source-bigquery/acceptance-test-docker.sh]: .sh files are out of the formatting scope, so I do it in an old fashion way :smile:

**Review by alafanechere — APPROVED** [review]
Thank you for this! 
Please log two additionals issues to add to the epic:
* Update the java-jdbc code connector to create these new files on new connectors
* Change execution logic of SAT to not require `acceptance.py` files

Before merging please update your branch: Connectors base is failing but the error in the logs is now fixed on master. Please wait for a successful connector base build afterward.

**Comment by DoNotPanicUA** [comment]
Update `source-jdbc-source` template #19130

**Comment by DoNotPanicUA** [comment]
@alafanechere 
#19133

File: airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json [hunk 4]
```diff
@@ -254,16 +240,14 @@
...
-                "default": "CDC",
                 "order": 0
               },
               "plugin": {
                 "type": "string",
                 "title": "Plugin",
                 "description": "A logical decoding plugin installed on the PostgreSQL server. The `pgoutput` plugin is used by default. If the replication table contains a lot of big jsonb values it is recommended to use `wal2json` plugin. Read more about <a href=\"https://docs.airbyte.com/integrations/sources/postgres#step-2-select-a-replication-plugin\">selecting replication plugins</a>.",
                 "enum": ["pgoutput", "wal2json"],
-                "default": "pgoutput",
+                "const": "pgoutput",
```
Comment (edgao) [inline airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json:250]: was this change intended? It seems to have removed the ability for users to switch to a different plugin https://github.com/airbytehq/oncall/issues/1133

Comment (edgao) [reply airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json:250]: submitted https://github.com/airbytehq/airbyte/pull/19985, if this passes SATs then I assume things are fine :shrug:

Comment (DoNotPanicUA) [reply airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json:250]: The SAT forced me to use `const` only.
